plugins/inputs/prometheus/kubernetes.go

```diff
@@ -0,0 +1,116 @@
+package prometheus
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/rest"
+)
+
+func start(p *Prometheus) error {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	watchlist := cache.NewListWatchFromClient(clientset.CoreV1().RESTClient(), "pods", v1.NamespaceAll, fields.Everything())
+	_, controller := cache.NewInformer(
+		watchlist,
+		&v1.Pod{},
+		time.Second*0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				pod := obj.(*v1.Pod)
+				registerPod(pod, p)
+			},
+			DeleteFunc: func(obj interface{}) {
+				pod := obj.(*v1.Pod)
+				unregisterPod(pod, p)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				podPod := oldObj.(*v1.Pod)
+				newPod := newObj.(*v1.Pod)
+				unregisterPod(podPod, p)
+				registerPod(newPod, p)
+			},
+		},
+	)
+
+	go controller.Run(wait.NeverStop)
+	return nil
+}
+
+func registerPod(pod *v1.Pod, p *Prometheus) {
+	url := scrapeURL(pod)
+	if url != nil {
+		log.Printf("Will scrape metrics from %v\n", *url)
```

Comment:
For these log at debug level: "`D! [inputs.prometheus] adding %q to scrape targets`"